packages/AI/Vectors/src/models/VectorBase.ts

```diff
@@ -0,0 +1,244 @@
+    }
+
+    protected async getRecordsByEntityID(entityID: number, recordIDs?: PrimaryKeyValueBase[]): Promise<BaseEntity[]> {
+        const rvResult = await this._runView.RunView({
+            EntityName: "Entities",
+            ExtraFilter: `ID = ${entityID}`,
+            ResultType: 'entity_object'
+        }, this.CurrentUser);
+
+        if(!rvResult.Success){
+            throw new Error(rvResult.ErrorMessage);
+        }
+        else if(rvResult.RowCount === 0){
+            throw new Error(`No Entity with ID=${entityID} not found`);
+        }
+
+        const entity: EntityEntity = rvResult.Results[0] as EntityEntity;
+        const rvResult2 = await this._runView.RunView({
+            EntityName: entity.Name,
+            ExtraFilter: recordIDs ? this.buildExtraFilter(recordIDs): undefined,
+            ResultType: 'entity_object'
+        }, this.CurrentUser);
+
+        if(!rvResult2.Success){
+            throw new Error(rvResult2.ErrorMessage);
+        }
+
+        return rvResult2.Results;
+    }
+
+    protected buildExtraFilter(keyValues: PrimaryKeyValueBase[]): string {
+        return keyValues.map((keyValue) => {
+            return keyValue.PrimaryKeyValues.map((keys: PrimaryKeyValue) => {
+                return `${keys.FieldName} = '${keys.Value}'`;
+            }).join(" AND ");
+        }).join("\n OR ");
+    }
+
+    protected async createEntityDocumentForEntity(entityID: number, vectorDatabase: VectorDatabaseEntity, AIModel: AIModelEntity): Promise<EntityDocumentEntity> {
+        const entity: EntityEntity = await this.runViewForSingleValue<EntityEntity>("Entities", `ID = ${entityID}`);
+        if(!entity){
+            LogError(`Entity with ID ${entityID} not found`);
+            return null;
+        }
+
+        const rvResult = await this._runView.RunView({
+            EntityName: "Entity Fields", 
+            ExtraFilter: `EntityID = ${entityID}`
+        }, this.CurrentUser);
+
+        if(!rvResult.Success){
+            LogError(rvResult.ErrorMessage);
+            return null;
+        }
+        else if(rvResult.RowCount === 0){
+            LogError(`No fields found for entity ${entity.Name}`);
+            return null;
+        }
+
+        const entityFields = rvResult.Results;
+        const EDTemplate: string = entityFields.map(entityField => {
+            return `${entityField.Name}: \$\{${entityField.Name}\}`;
+        }).join(" ");
+
+        const md: Metadata = new Metadata();
+        const entityDocument: EntityDocumentEntity = await md.GetEntityObject<EntityDocumentEntity>("Entity Documents");
+        entityDocument.NewRecord();
+        entityDocument.Set("Name", `Default duplicate record Entity Document for the ${entity.Name} entity using vector database ${vectorDatabase.Name} and AI Model ${AIModel.Name}`);
+        entityDocument.Set("EntityID", entityID);
+        entityDocument.Set("TypeID", 9); //hardcoded, but we know that 9 is the ID for Record Duplicates
```

Comment:
Might want to consider adding a `constants.ts` with `export RECORD_DUPLICATES_TYPE_ID=9;` just for easier maintainability down the road